cmd/proxy/actions/filter.go

```diff
@@ -30,6 +38,18 @@ func newFilterMiddleware(mf *module.Filter) buffalo.MiddlewareFunc {
 			// i.e. list requests path is like /{module:.+}/@v/list with no version parameter
 			version, _ := paths.GetVersion(c)
 
+			if validatorHook, ok := env.ValidatorHook(); ok && version != "" {
+				valid, err := validate(validatorHook, mod, version)
+				if err != nil {
+					entry.SystemErr(err)
+					return c.Render(http.StatusInternalServerError, nil)
+				}
+
+				if !valid {
+					return c.Render(http.StatusForbidden, nil)
```

Comment:
should this return whatever status code the webhook returned?

Reply:
I'd prefer to keep the interface between the proxy and the hook clear, stating that we only expect 200 and 403 as results and the proxy behave according to those two results (looking at the valid result). 403 is incidentally the result the proxy returns after a 403 from the hook, but I would avoid to propagate the result back. As it is now, every other status code raises a (logged) error because we expect only those two values from the hook